spec/unit/resources/sh_cluster.rb

```diff
@@ -0,0 +1,90 @@
+# coding: UTF-8
+
+require_relative '../spec_helper'
+
+describe 'cerner_splunk::_update_shcluster' do
+  subject do
+    runner = ChefSpec::Runner.new(step_into: ['cerner_splunk_sh_cluster']) do |node|
+      node.set['splunk']['config']['clusters'] = ['cerner_splunk/cluster']
+      node.set['splunk']['cmd'] = '/opt/splunk/bin/splunk'
+      node.set['splunk']['config']['sh_member_action'] = action
+      node.set['splunk']['package']['base_name'] = 'base_name'
+      node.set['splunk']['package']['download_group'] = 'download_group'
+      node.run_state['cerner_splunk'] = {}
+      node.run_state['cerner_splunk']['admin-password'] = 'admin_password'
+    end
+    runner.converge('cerner_splunk::_install', 'cerner_splunk::captain', described_recipe)
+  end
+
+  let(:cluster_config) do
+    {
+      'receivers' => ['33.33.33.20'],
+      'license_uri' => nil,
+      'receiver_settings' => {
+        'splunktcp' => {
+          'port' => '9997'
+        }
+      },
+      'indexes' => 'cerner_splunk/indexes',
+      'sh_cluster' => [
+        'https://33.33.33.15:8089',
+        'https://33.33.33.17:8089'
+      ]
+    }
+  end
+
+  let(:existing_member) { nil }
+  let(:action) { nil }
+
+  before do
+    allow(Chef::DataBagItem).to receive(:load).with('cerner_splunk', 'cluster').and_return(cluster_config)
+    allow(Chef::DataBagItem).to receive(:load).with('cerner_splunk', 'indexes').and_return({})
+    stub_command('/opt/splunk/bin/splunk list shcluster-members -auth admin:admin_password | grep 127.0.0.1').and_return(existing_member)
+    stub_command('/opt/splunk/bin/splunk list shcluster-members -auth admin:admin_password | grep is_captain:1').and_return(existing_member)
+  end
+
+  after do
+    CernerSplunk.reset
+  end
+
+  context 'when the captain needs to be assgined' do
```

Comment:
Made some updates to this test. It should fix it. 61dc0e035f9f931905c14c58001a6f80a94475d9